swayfmt/src/utils/map/newline.rs

```diff
@@ -149,18 +143,65 @@ fn add_newlines(
         .skip(1)
         .zip(formatted_newline_spans.iter().skip(1))
     {
-        if let Some(newline_sequence) = get_newline_sequence_between_spans(
-            previous_unformatted_newline_span,
-            unformatted_newline_span,
-            &newline_map,
-        ) {
-            offset += insert_after_span(
-                previous_formatted_newline_span,
-                newline_sequence,
-                offset,
-                formatted_code,
-                newline_threshold,
-            )?;
+        if previous_unformatted_newline_span.end < unformatted_newline_span.start {
+            let snippet = &unformatted_code
+                [previous_unformatted_newline_span.end..unformatted_newline_span.start];
+
+            let start = previous_unformatted_newline_span.end;
+
+            // Here, we will try to insert newlines that occur before and after comments.
+            // The reason we do this is because comments aren't a part of the AST, and so they aren't
+            // collected as leaf spans - they simply exist between whitespaces.
+            if let Some(start_first_comment) = snippet.find("//") {
```

Comment:
Hmm does this check with `//` breaks anything related to the comments starting with `/*`?

Reply:
Good catch!

I added some naive checks for now that handles catching this, but in future this should probably be extended as the formatter matures to handle other comment formats as well.